src/imlib.c

```diff
@@ -523,9 +523,15 @@ void feh_reload_image(winwidget w, int resize, int force_new)
 	 * However, if --reload is used (force_new == 0), we want to continue if
 	 * the new image cannot be loaded, so we must not free the old image yet.
 	 */
-	if (force_new)
+	if (force_new) {
 		winwidget_free_image(w);
 
+		/* Flush cache, so it will surely not be in the way. */
+		int cache_size = imlib_get_cache_size();
+		imlib_set_cache_size(0);
```

Comment:
Is this necessary? winwidget_free_image calls imlib_free_image_and_decache internally, which according to https://git.enlightenment.org/legacy/imlib2.git/tree/src/lib/api.c#n1392 ensures that it's no longer present in the cache.

Reply:
Isn't `winwidget_free_image()` calls `imlib_free_image()` only?

But maybe I should call `winwidget_free_image_and_decache()` or something instead of flushing the whole cache.

Reply:
Oh, you're right. I mixed up `winwidget_free_image` with `winwidget_free_image_and_decache` when looking up the code paths.

> But maybe I should call `winwidget_free_image_and_decache()` or something instead of flushing the whole cache.

IMHO that's a cleaner solution, yeah. Apart from that, the PR is looking good, thanks :)